### DISCUSSION TITLE
Expose NavLink isActive/isPending logic

### DISCUSSION THREAD

**Discussion by tpict** [remix-run/react-router#10224] [Proposals]
Credit to @afzalsayed96 and their original PR for this proposal: https://github.com/remix-run/react-router/pull/10183

Being able to check "does the given path match the current location", subject to conditions like "is relative", "is case sensitive" etc, has broad applications. The `NavLink` component contains an implementation of that check, but it's not exposed to users of react-router. Personally I'm not comfortable copy-pasting it into my codebase, because there's enough complexity that it might wind up blocking attempts to upgrade to future react-router releases.

It would be great if this logic was exposed as a function or hook in the library. This would also resolve [issues like this](https://github.com/remix-run/react-router/discussions/9840) without adding more complexity to `NavLink`.

**Comment by willhoney7** [comment]
Would love to see this addressed!

**Comment by MoSattler** [comment]
Me as well! An example is when using Tailwind Catalyst Sidebars, which rely on a `current` prop, and cannot easily use NavLink!
